crates/bin/output_segment/src/main.rs

```diff
@@ -0,0 +1,82 @@
+use std::fs;
+use std::path::PathBuf;
+
+use cairo_vm::hint_processor::hint_processor_utils::felt_to_usize;
+use cairo_vm::types::layout_name::LayoutName;
+use cairo_vm::vm::runners::cairo_pie::CairoPie;
+use cairo_vm::Felt252;
+use clap::Parser;
+use prove_block::{debug_prove_error, get_memory_segment, prove_block, DEFAULT_COMPILED_OS};
+
+#[derive(Parser, Debug)]
+struct Args {
+    /// Pie-zip to compare.
+    #[arg(long = "pie-path")]
+    zip_pie_path: PathBuf,
+
+    /// RPC endpoint to use for fact fetching
+    #[arg(long = "rpc-provider", default_value = "http://localhost:9545")]
+    rpc_provider: String,
+}
+
+fn init_logging() {
+    env_logger::builder()
+        .filter_level(log::LevelFilter::Debug)
+        .format_timestamp(None)
+        .try_init()
+        .expect("Failed to configure env_logger");
+}
+
+#[tokio::main]
+async fn main() {
+    init_logging();
+
+    let args = Args::parse();
+
+    let zip_pie_path = args.zip_pie_path;
+    let endpoint = args.rpc_provider;
+
+    let reference_pie_bytes = fs::read(zip_pie_path).unwrap();
+    let reference_pie = CairoPie::from_bytes(&reference_pie_bytes).expect("reference PIE");
+    reference_pie.run_validity_checks().expect("Valid reference PIE");
+
+    let block_number: u64 =
+        felt_to_usize(&get_pie_block_number(&reference_pie)).unwrap().try_into().expect("Block number is too big");
+
+    log::info!("Runnin SNOS for block number: {}", block_number);
+
+    let (snos_pie, _snos_output) =
+        prove_block(DEFAULT_COMPILED_OS, block_number, &endpoint, LayoutName::all_cairo, true)
+            .await
+            .map_err(debug_prove_error)
+            .expect("OS generate Cairo PIE");
+
+    snos_pie.run_validity_checks().expect("Valid SNOS PIE");
+
+    let output_segment_index = 2;
```

Comment:
Is there any reference you can add about this?

Reply:
You wanted, you have it 😎